drivers/SmartThings/jbl/src/discovery_mdns.lua

```diff
@@ -0,0 +1,154 @@
+    for _, text_item in pairs(answer_text or {}) do
+        table.insert(text_list, text_item)
+    end
+    return text_list
+end
+
+local function filter_response_by_servie_name(service_type, domain, discovery_responses)
+    local filtered_responses = {
+      answers = {},
+      found = {}
+    }
+
+    for _, answer in pairs(discovery_responses.answers or {}) do
+      table.insert(filtered_responses.answers, answer)
+    end
+
+    for _, additional in pairs(discovery_responses.additional or {}) do
+      table.insert(filtered_responses.answers, additional)
+    end
+
+    for _, found in pairs(discovery_responses.found or {}) do
+      if found.service_info.service_type == service_type then
+        table.insert(filtered_responses.found, found)
+      end
+    end
+
+    return filtered_responses
+end
+
+local function insert_dni_ip_from_answers(driver, filtered_responses, target_table)
+    for _, answer in pairs(filtered_responses.answers) do
+        local dni, ip
+        log.info("answer_name, arecod = " .. tostring(answer.name) .. ", " .. tostring(answer.kind.ARecord))
+
+        if answer.kind.ARecord ~= nil then
+            ip = answer.kind.ARecord.ipv4
+        end
+
+        if ip ~= nil then
+            if dni == nil then
+                dni = driver.discovery_helper.get_dni(driver, ip, filtered_responses)
+            end
+    
+            if dni ~= nil then
+                target_table[dni] = ip
+            end
+        end
+    end
+end
+
+local function insert_dni_ip_from_found(driver, filtered_responses, target_table)
+    for _, found in pairs(filtered_responses.found) do
+        local dni, ip
+        log.info("found_name = " .. tostring(found.service_info.service_type))
+        if found.host_info.address ~= nil and net_utils.validate_ipv4_string(found.host_info.address) then
+            log.info("ip = " .. tostring(found.host_info.address))
+            ip = found.host_info.address
+        end
+
+        if ip ~= nil then
+            if dni == nil then
+                dni = driver.discovery_helper.get_dni(driver, ip, filtered_responses)
+            end
+
+            if dni ~= nil then
+                target_table[dni] = ip
+            end
+        end
+    end
+end
+
+local function get_dni_ip_table_from_mdns_responses(driver, service_type, domain, discovery_responses)
+    local dni_ip_table = {}
+
+    local filtered_responses = filter_response_by_servie_name(service_type, domain, discovery_responses)
+
+    insert_dni_ip_from_answers(driver, filtered_responses, dni_ip_table)
+    insert_dni_ip_from_found(driver, filtered_responses, dni_ip_table)
+
+    return dni_ip_table
+end
+
+function discovery_mdns.find_ip_table_by_mdns(driver)
+    log.info("discovery_mdns.find_device_ips")
+
+    local service_type, domain = driver.discovery_helper.get_service_type_and_domain()
+    local discovery_responses = mdns.discover(service_type, domain) or {answers = {}, additional = {}}
```

Comment:
This is not the correct return type structure for the `mdns.discover` API; this was the return behavior in an early version of the API but it change around hubcore 46.

The final version of the API that is available to production drivers can be found here: https://developer.smartthings.com/docs/edge-device-drivers/mdns.html